### DISCUSSION TITLE
is it possible to return JSON results if Button 2 is pressed

### DISCUSSION THREAD

**Discussion by LtCmdrScott** [swiftDialog/swiftDialog#358] [General]
I have a user selectable list of backup locations and two buttons...Button 0 is the "Backup" button and button 1 is the "Restore" option. ...if the user presses "Restore", SW treats it as an "Esc" button press and doesn't give me the results of the JSON choices.  Is there a way have button 1 return JSON results, or if not, is there a way to have additional buttons to add to the dialog?

**Comment by bartreardon** [comment]
no and not yet. button 2 is a "cancel everything and get out" button by design although there's been a couple of requests for alternate button behaviours. There's currently no open feature requests for that though. If you can open a FR describing the behaviour then I'll look into adding it to the next release.

(apart from acting as a reminder to me to look into it becasue I'll forget otherwise, if you open the FR then I'll know who to give credit to in the release notes 🙂 )
